packages/nouns-contracts/test/foundry/NounsDescriptorV3.t.sol

```diff
@@ -0,0 +1,742 @@
+
+        ISVGRenderer.Part[] memory parts = descriptor.getPartsForSeed(INounsSeeder.Seed(0, 0, 0, 0, 0));
+
+        assertEq(parts[0].image, 'the body');
+        assertEq(parts[0].palette, 'the palette');
+
+        assertEq(parts[1].image, 'the accessory');
+        assertEq(parts[1].palette, 'the palette');
+
+        assertEq(parts[2].image, 'the head');
+        assertEq(parts[2].palette, 'the palette');
+
+        assertEq(parts[3].image, 'the glasses');
+        assertEq(parts[3].palette, 'the palette');
+    }
+
+    function _makeArtGettersNotRevert() internal {
+        vm.mockCall(address(art), abi.encodeWithSelector(INounsArt.backgroundCount.selector), abi.encode(123));
+        vm.mockCall(address(art), abi.encodeWithSelector(INounsArt.backgrounds.selector), abi.encode('return value'));
+        vm.mockCall(address(art), abi.encodeWithSelector(INounsArt.bodies.selector), abi.encode('return value'));
+        vm.mockCall(address(art), abi.encodeWithSelector(INounsArt.accessories.selector), abi.encode('return value'));
+        vm.mockCall(address(art), abi.encodeWithSelector(INounsArt.heads.selector), abi.encode('return value'));
+        vm.mockCall(address(art), abi.encodeWithSelector(INounsArt.glasses.selector), abi.encode('return value'));
+        vm.mockCall(address(art), abi.encodeWithSelector(INounsArt.palettes.selector), abi.encode('return value'));
+    }
+}
+
+contract NounsDescriptorV3WithRealArtTest is DeployUtils {
+    using strings for *;
+    using stdJson for string;
+    using Base64 for string;
+
+    NounsDescriptorV3 descriptor;
+
+    uint256 glassesIndex = 1;
+
+    function setUp() public {
+        descriptor = _deployAndPopulateV3();
+    }
+
+    function testGeneratesValidTokenURI() public {
+        string memory uri = descriptor.tokenURI(
+            0,
+            INounsSeeder.Seed({ background: 0, body: 0, accessory: 0, head: 0, glasses: 0 })
+        );
+
+        string memory json = string(removeDataTypePrefix(uri).decode());
+        string memory imageDecoded = string(removeDataTypePrefix(json.readString('.image')).decode());
+        strings.slice memory imageSlice = imageDecoded.toSlice();
+
+        assertEq(json.readString('.name'), 'Noun 0');
+        assertEq(json.readString('.description'), 'Noun 0 is a member of the Nouns DAO');
+        assertEq(bytes(imageDecoded).length, 6849);
+        assertTrue(
+            imageSlice.startsWith(
+                '<svg width="320" height="320" viewBox="0 0 320 320" xmlns="http://www.w3.org/2000/svg" shape-rendering="crispEdges">'
+                    .toSlice()
+            )
+        );
+        assertTrue(
+            imageSlice.endsWith(
+                '<rect width="60" height="10" x="100" y="160" fill="#ff638d" /><rect width="60" height="10" x="170" y="160" fill="#ff638d" /></svg>'
+                    .toSlice()
+            )
+        );
+    }
+
+    function testUpdateGlasssWorks() public {
```

Comment:
@EricDevito I'm wondering now that you added this test code, why not include in it the same checks you have in the typescript test file, and then just remove the typescript test?

we are generally trying to be foundry maxis as much as we can.

wdyt?